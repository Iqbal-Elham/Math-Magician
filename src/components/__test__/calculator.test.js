import { render, screen } from '@testing-library/react';
import Calculator from '../calculator'
import renderer from 'react-test-renderer';

describe("Calculator", () => {
    it('should return the length of the buttons', () => {
        render(<Calculator />);
      
        const buttons = screen.getAllByRole('button');
      
        expect(buttons).toHaveLength(19);
      });

      it('Render the user interaction', () => {
        const tree = renderer.create(<Calculator />).toJSON();
        expect(tree).toMatchSnapshot();
      })
})