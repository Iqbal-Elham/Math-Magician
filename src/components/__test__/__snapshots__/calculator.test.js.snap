// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Render the user interaction 1`] = `
<div
  className="calc-container"
>
  <div
    className="result-screen"
  >
    0
  </div>
  <div
    className="buttons"
  >
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      /
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      *
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="zero-btn"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="gray-btn"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="orange-btn"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
